frontend/src/components/pages/CampCreation/index.tsx

```diff
@@ -159,87 +170,254 @@ const CampCreationPage = (): React.ReactElement => {
     });
   };
 
+  // The edit-camp route will have an id to identify the camp currently in draft state
+  const { id: editCampId } = useParams<{ id: string }>();
+  const [isLoadedCampData, setIsLoadedCampData] = useState<boolean>(false);
+
+  // This useEffect fetches the current state of the draft camp if we are editing camp
+  useEffect(() => {
+    const setInitialEditState = async () => {
+      // retrieve the current state of the camp to edit
+      const editCamp = await CampsAPIClient.getCampById(editCampId);
+
+      if (editCamp) {
+        // Schedule Sessions need the data in a certain format. We convert the array of dates stored in the backend
+        // to this array of CreateCampSessions so that the sessions can be viewed/edited/deleted
+        const currentCampSessions: CreateCampSession[] = editCamp.campSessions.map(
+          (cs) => {
+            const dates = cs.dates
+              .map((date) => new Date(date))
+              .sort((a, b) => a.getTime() - b.getTime());
+            const currentWeekDays = getSelectedWeekDaysFromDates(dates);
+            return {
+              startDate: dates[0],
+              endDate: dates[dates.length - 1],
+              dates,
+              selectedWeekDays: currentWeekDays,
+            };
+          },
+        );
+        setScheduledSessions(currentCampSessions);
+
+        // We set the customQuestions to be the current questions stored on the backend so the admin
+        // can view and edit the questions in the ui smoothly
+        const currentFormQuestions: CreateFormQuestion[] = editCamp.formQuestions.map(
+          (fq) => {
+            return {
+              question: fq.question,
+              type: fq.type,
+              category: fq.category,
+              required: fq.required,
+              description: fq.description,
+              options: fq.options,
+            };
+          },
+        );
+        setCustomQuestions(currentFormQuestions);
+
+        // Set basic camp details
+        setCampName(editCamp.name);
+        setCampDescription(editCamp.description);
+        setDailyCampFee(editCamp.fee);
+        setStartTime(editCamp.startTime);
+        setEndTime(editCamp.endTime);
+        setAgeLower(editCamp.ageLower);
+        setAgeUpper(editCamp.ageUpper);
+        setOffersEDLP(
+          editCamp.earlyDropoff.length !== 0 ||
+            editCamp.latePickup.length !== 0,
+        );
+        setEarliestDropOffTime(editCamp.earlyDropoff);
+        setLatestPickUpTime(editCamp.latePickup);
+        setPriceEDLP(editCamp.pickupFee);
+        setAddresLine1(editCamp.location.streetAddress1);
+        setAddresLine2(editCamp.location.streetAddress2 ?? "");
+        setCity(editCamp.location.city);
+        setProvince(editCamp.location.province);
+        setPostalCode(editCamp.location.postalCode);
+        setCampImageURL(editCamp.campPhotoUrl);
```

Comment:
would we be safe to make this an actual camp object eg. `CampResponse`? (I'm hoping enough of the flow is filled in to the point that this would be a useful thing to do, and reduce some of the repetitiveness of the getters/setters)

Reply:
it would but i didn't want to spend time on it- don't think it's worth making the code beautiful at this point honestly